processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -159,12 +162,14 @@
     final boolean includeTimestamp = GroupByStrategyV2.getUniversalTimestamp(query) == null;
 
     final ThreadLocal<Row> columnSelectorRow = new ThreadLocal<>();
-    final ColumnSelectorFactory columnSelectorFactory = query.getVirtualColumns().wrap(
-        RowBasedColumnSelectorFactory.create(
-            columnSelectorRow,
-            rawInputRowSignature
-        )
+
+    ColumnSelectorFactory columnSelectorFactory = RowBasedColumnSelectorFactory.create(
+        columnSelectorRow,
+        rawInputRowSignature
     );
+    if (useVirtualizedColumnSelectorFactory) {
```

Comment:
Things would work without this change but not in most optimal way.

`VirtualizedColumnSelectorFactory` always works by the "row" based methods (that existed before) , however point of this PR is to use "column" based version of those methods wherever possible (e.g. "column" is not available when processing query on IncrementalIndex or processing outer query of a groupBy and outer query has virtual columns).
Without this change, all groupBy query processing would blindly use row based methods in VirtualColumn and discard more efficient column based methods (even if they were implemented) , this change is making sure that we use row based methods only when necessary during groupBy query processing.
there are unit tests introduced in this PR which would fail without this change, to ensure that things stay this way.

Reply:
Thanks for the explanation, I'll have a closer look to have a better understanding what is going on here. Can you capture this in a code comment or link to this github comment?